lib/presentation/routes/extension.dart

```diff
@@ -2,17 +2,35 @@ part of 'routes.dart';
 
 extension XBuildContext on BuildContext {
   void goToScreen({required RouteArg arg}) {
+    AnalyticsService().logEvent(
+      'go_to_screen',
+      <String, dynamic>{
+        'screen': arg.parsedPath,
+      },
+    );
```

Comment:
why doing the same thing in 2 different ways. you can logging here manually. but you have also added observer below. doesn't make much sense. remove all the logs from here. And create your own RouteAnalyticsObserverObserver, instead of using firebase's FirebaseAnalyticsObserver

Reply:
Will talk about that.

Reply:
yeah 😄